### PR TITLE
[Balloon] Update public documentation

### PR DESCRIPTION
**Pull request by georgepisaltu**
## Reason for This PR

The balloon public documentation was not clear enough about our recommendations for cases where the guest driver is not cooperative.

## Description of Changes

Added user recommendations in the balloon public documentation for the case where the memory restrictions imposed on the VM through the balloon device are broken.

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [ ] Any newly added `unsafe` code is properly documented.
- [ ] Any API changes are reflected in `firecracker/swagger.yaml`.
- [ ] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

### CODE REVIEW

File: docs/ballooning.md [hunk 1]
```diff
@@ -61,11 +60,19 @@ memory usage, they can be broken by a compromised driver in the guest. The
...
 uses all of the memory it was given at boot even if the balloon was used to
-restrict the amount of memory available to the guest.
+restrict the amount of memory available to the guest. It is also the users'
+responsibility to monitor the memory consumption of the VM and, in case
+unexpected increases in memory usage are observed, we recommend the following
+options:
+* through the snapshot capabilities, migrate the VM to a machine with higher
+memory availability;
+* kill the Firecracker process that exceeds memory restrictions;
+* enable swap with a sufficient amount of memory to handle the memory demand;
```
Comment (sandreim) [inline docs/ballooning.md:70]: I would also mention the cost of doing so.

File: docs/ballooning.md [hunk 2]
```diff
@@ -61,11 +60,19 @@ memory usage, they can be broken by a compromised driver in the guest. The
 balloon device operates on a best effort model and users should always ensure
 the host is prepared to handle a situation in which the Firecracker process
 uses all of the memory it was given at boot even if the balloon was used to
-restrict the amount of memory available to the guest.
+restrict the amount of memory available to the guest. It is also the users'
+responsibility to monitor the memory consumption of the VM and, in case
+unexpected increases in memory usage are observed, we recommend the following
+options:
+* through the snapshot capabilities, migrate the VM to a machine with higher
```
Comment (sandreim) [inline docs/ballooning.md:67]: I would rephrase to moving the VM at the cost of disrupting the workload.

Comment (georgepisaltu) [reply docs/ballooning.md:67]: Done.

File: docs/ballooning.md (hunk 1 above)
Comment (georgepisaltu) [reply docs/ballooning.md:70]: Done.

**Review by dianpopa — APPROVED** [review]

**Review by sandreim — APPROVED** [review]
